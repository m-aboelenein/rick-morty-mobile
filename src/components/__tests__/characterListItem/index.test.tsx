import * as React from 'react';
import CharacterListItem from '../../characterListItem';
import renderer from 'react-test-renderer';
import {render, fireEvent} from '@testing-library/react-native';

const mockOnPressCharacter = jest.fn();

it('renders a character item', () => {
  const tree = renderer.create(
    <CharacterListItem
      onPressCharacter={mockOnPressCharacter}
      character={{
        id: 1,
        name: 'Rick Sanchez',
        image: 'https://rickandmortyapi.com/api/character/avatar/1.jpeg',
      }}
    />,
  );
  expect(tree).toMatchSnapshot();
});

it('navigates to character details', () => {
  const {getByTestId} = render(
    <CharacterListItem
      onPressCharacter={mockOnPressCharacter}
      character={{
        id: 1,
        name: 'Rick Sanchez',
        image: 'https://rickandmortyapi.com/api/character/avatar/1.jpeg',
      }}
    />,
  );

  fireEvent.press(getByTestId('character-item'));
  expect(mockOnPressCharacter).toHaveBeenCalled();
});
