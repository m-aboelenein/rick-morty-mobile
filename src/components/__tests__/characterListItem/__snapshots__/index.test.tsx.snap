// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a character item 1`] = `
<View
  accessible={true}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "borderRadius": 20,
      "flexDirection": "row",
      "marginVertical": 8,
      "minHeight": 80,
      "paddingHorizontal": 16,
    }
  }
  testID="character-item"
>
  <Image
    source={
      Object {
        "uri": "https://rickandmortyapi.com/api/character/avatar/1.jpeg",
      }
    }
    style={
      Object {
        "borderRadius": 35,
        "height": 70,
        "marginRight": 16,
        "resizeMode": "cover",
        "width": 70,
      }
    }
  />
  <Text
    numberOfLines={1}
    style={
      Object {
        "color": "black",
        "flex": 1,
        "fontSize": 16,
      }
    }
  >
    Rick Sanchez
  </Text>
</View>
`;
